### PR TITLE
Disable check_to_wkt_trim_compatible for GEOS >= 3.13.0

### PR DESCRIPTION
**Pull request by mwtoews**
This is a follow-up to #1907 that considers upstream changes for upcoming GEOS 3.13.0 in https://github.com/libgeos/geos/pull/973

It also fixes a related regression with tests against GEOS main branch, where small non-zero coordinates are shown with scientific notation (with trim enabled) rather than misrepresenting them as zero.

### CODE REVIEW

**Review by caspervdw — APPROVED** [review]

File: shapely/tests/geometry/test_format.py [hunk 1]
```diff
@@ -93,7 +93,12 @@ def test_format_polygon():
     )
 
     # 'g' format varies depending on GEOS version
-    if geos_version < (3, 10, 0):
+    if geos_version >= (3, 13, 0):
+        expected_2G = (
+            "POLYGON ((10 0, 7.07 -7.07, 6.12E-16 -10, -7.07 -7.07, "
+            "-10 -1.22E-15, -7.07 7.07, -1.84E-15 10, 7.07 7.07, 10 0))"
```
Comment (mwtoews) [inline shapely/tests/geometry/test_format.py:99]: FYI, I'm a bit uneasy (and pedantic) about this one an and considering an upstream change to zap the near-zero values to zero, so this test might change once more before GEOS 3.13 is finalized.

Comment (mwtoews) [reply shapely/tests/geometry/test_format.py:99]: xref https://github.com/libgeos/geos/pull/978
